### ISSUE TITLE
Method @ Annotations Mishandling In Method Header Generation

### ISSUE THREAD

**Issue by gussamer** [HugoOM/sfdx_autoheader#23]
Hello I love this extension but I've got a little issue with the method generation.

Now this could be considered two issues and I can split them apart at your discretion.

In apex I prefer to write my method signatures which include @ annotations (@TestVisible, @IsTest, etc...) on the same line as seen below.

    /**
    * @description 
    * @author Angus | 9/13/2019 
    * @return @IsTest 
    **/
    @IsTest private static void test1(){

    }

Now we can see the issue is that @IsTest has been picked up in the return value instead of the correct return value void. I skimmed the code and see that you are determining the return value by finding the first value not contained in a list of apex reserved words and that list does not contain any of the @ annotations.  I assume by adding the @ annotations to this list it would fix my issue but I haven't had time to rig that up and test it. I'll try and find some time to test this myself and if successful would try and get a pull request together. But if you can get to it before I can it would be much appreciated.

So for the second part of this issue is that upon discovering the above I tried generating method headers with the annotations the more widely used format but that also had an issue.

    @IsTest
    /**
    * @description 
    * @author Angus | 9/13/2019 
    * @return void 
    **/
    private static void test2(){

    }

I'm assuming this is a result of inserting immediately above the method signature without awareness of the annotation. This I doubt will be as easy to fix as the above as there is a lot more possible variance in detecting how far up above the method signature the comment needs to be inserted.

**Comment by HugoOM** [comment]
@gussamer Thanks for the feedback, I didn't think about the first one and never ran into it since I always  use the second syntax :) 

I have yet to extend the test coverage to the method header generation, but I'll make that a test case in the future. 

The second issue you raised has been bothering me for a while too so I might as well get to it! 

I'll try to get something out for both of those shortly.

Thanks again for taking the time to report those and for your interest in the extension, it's much appreciated!

Hugo

**Comment by gussamer** [comment]
@HugoOM 

Understandable I don't see many people using the single line syntax so I assumed it was never considered.

And I figured inserting below the annotations was known but was tabled for later due to complexity and that it's really just annoying rather than functionally broken.

No thank you apex comments have been a pain point for me for quite awhile until I found your extension so I'm happy to contribute any way I can.

Angus

**Comment by HugoOM** [comment]
Hi @gussamer, 

Glad the extension proves of use to you :) 

I have a fix in place for both aforementioned issues, which I'll be testing locally for a few days before releasing. 

It covers most scenarios for "inline" annotations: 

```java
    /**
    * @description 
    * @author hmonette@deloitte.ca | 2019-09-14 
    * @param anInteger 
    * @return void 
    **/
    @isTest @ http private void testMethod2(Integer anInteger) {}
```

Split line (single or multiple)
```java
    /**
    * @description 
    * @author hmonette@deloitte.ca | 2019-09-14 
    * @param anInteger 
    * @return void 
    **/
    @isTest 
    @http
    private void testMethod2(Integer anInteger) {}
```

And mixed, for those of us who are funkier

```java
    /**
    * @description 
    * @author hmonette@deloitte.ca | 2019-09-14 
    * @param anInteger 
    * @return void 
    **/
    @         http 
    @ isTest private void testMethod2(Integer anInteger) {}
```

I just learned today though that any whitespace character ```\s*``` is considered valid by the Apex compiler ... That is to say that aberrations such as

```java 
@


isTest private void testMethod2(Integer anInteger) {}
```

... Are actually considered valid Apex.

I have made a decision, at least for now, not to cater for line breaks between the ```@``` and the actual annotation declaration. 

As shown above, any "on-line" whitespace character will be processed properly by the extension, but I feel that this specific edge case is different, because:

- It would take a separate and relatively complex logic to handle which would in turn further complexify the code base.
- This is not a use case that I see occuring often enough to justify the effort and added complexity.
- The larger goal of this extension is to promote code quality, and nobody should ever have to witness such horrific code. In my opinion, it shouldn't even be valid Apex in the first place and I don't feel like legitimizing its use through the extension is the way to go.
- If someone really wants it that way, they can always go through the motions. The extension will still generate a header in such case, and the @return argument can simply  be manually corrected, as is the case currently.

Please feel free to chime in if you have a diverging opinion, any question or comments.

Otherwise, I'll be updating/closing this issue once I release the fixes publicly.

Thanks again,
Hugo

**Comment by HugoOM** [comment]
Resolved in #24.

Thanks!
